src/group_theory/nilpotent.lean

```diff
@@ -482,3 +482,21 @@ begin
   calc derived_series G n ≤ lower_central_series G n : derived_le_lower_central n
     ... = ⊥ : hn
 end
+
+section normalizer_condition
+
+/-- Every proper subgroup `H` of `G` is a proper normal subgroup of the normalizer of `H` in `G`. -/
+def normalizer_condition (G : Type*) [group G] :=
+  ∀ (H : subgroup G), H < ⊤ → H < normalizer H
+
+/-- Alternative phrasing of the normalizer condition: Only the full group is self-normalizing.
+This may be easier to work with, as it avoids inequalities or negations.  -/
+lemma normalizer_condition_iff_only_full_group_self_normalizing :
+  normalizer_condition G ↔ ∀ (H : subgroup G), H.normalizer = H → H = ⊤ :=
+begin
+  apply forall_congr, intro H,
+  simp [normalizer_condition, lt_top_iff_ne_top, lt_iff_le_and_ne, le_normalizer],
+  tauto!,
```

Comment:
This is a non-terminal `simp`. Is that ok, given that it's followed by another automatic tactic?

Reply:
It probably won't cause any issues, but why not `squeeze_simp` it to a `simp only` unless the output is ridiculously long you don't lose too much by doing so, and the resulting proof will likely be faster anyway

Reply:
Thanks! Indeed nice this way.